src/FleetFlow.Api/Controllers/LocationsController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using FleetFlow.Service.DTOs;
+using FleetFlow.Service.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace FleetFlow.Api.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class LocationsController : ControllerBase
+    {
+        private readonly IlocationService locationService;
+        public LocationsController(IlocationService locationService)
+        {
+            this.locationService = locationService;
+        }
+
+        /// <summary>
+        /// Get all location
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public async ValueTask<IActionResult> GetAllAsync()
+        {
+            var locations = await locationService.GetAllAsync();
+            return Ok(locations);
+        }
+
+        /// <summary>
+        /// Get by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("Id")]
```

Comment:
In RESTful API naming conventions, it is recommended to use lowercase letters for endpoint urls. So, if you are using HttpGet("Id") in your code, it would be better to change it to HttpGet("id").

Using lowercase letters helps in making the endpoint URLs more readable, consistent, and easier to maintain. This practice also aligns with the convention of using lowercase letters for URL paths, which is widely adopted in web development.

Reply:
Additional Helper can be added to do it better using REGEX